src/app/link-list/link-list.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit, ApplicationRef } from '@angular/core';
+import { LinkService } from '../link.service';
+
+import { Observable, Subject} from 'rxjs';
+import { debounceTime, distinctUntilChanged, switchMap } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-link-list',
+  templateUrl: './link-list.component.html',
+  styleUrls: ['./link-list.component.css']
+})
+export class LinkListComponent implements OnInit {
+
+  constructor(private linkService: LinkService, private app: ApplicationRef) { }
+
+  links$: Observable<string[]>;
+  private searchTerms = new Subject();
+
+  search(term: string): void {
+    this.searchTerms.next(term);
+  }
+
+  ngOnInit(): void {
+    this.links$ = this.searchTerms.pipe(
+        distinctUntilChanged(),
+        debounceTime(300),
+        switchMap((term: string) => this.linkService.filterLinks(term))
+    );
+
+    this.linkService.invokeSearch.subscribe(() => {
```

Comment:
Not sure if I understand this part, what are you thing to achieve by this. Are you trying a reset the search box whenever a new link got added?

Reply:
Since there is a delay between receiving the links from the content script and the initial loading of the page, we need to manually invoke a search once the links have been loaded and remind angular to reload it's contents once the links are ready. We only need to do this once, when the links are first received because every other time that we require a reload is after user interaction which angular handles on its own. (I've added a comment to clarify this.)

Reply:
Actually, with the custom filter we aren't going to need the event emitter, we can just call app.tick() angular from the linkService